manticore/wasm/structure.py

```diff
@@ -488,8 +523,60 @@ class MemInst:
     https://www.w3.org/TR/wasm-core-1/#memory-instances%E2%91%A0
     """
 
-    data: typing.List[int]  #: The backing array for this memory
+    _published_events = {"write_memory", "read_memory"}
```

Comment:
I'm surprised `mypy` isn't unhappy. Usually it yells at me for missing class fields with type annotations.

Should be something like 

```suggestion
    _published_events: typing.Set[str] = {"write_memory", "read_memory"}
```

Reply:
It's because it inherits from Eventful, which already specifies the type.
`_published_events: Set[str] = set()`